controllers/networking-calico/pkg/controller/actuator_delete.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+
+	resourcemanager "github.com/gardener/gardener-resource-manager/pkg/manager"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+)
+
+// Delete implements Network.Actuator.
+func (a *actuator) Delete(ctx context.Context, network *extensionsv1alpha1.Network) error {
+	if err := resourcemanager.NewManagedResource(a.client).
+		WithNamespacedName(network.Namespace, network.Name).
+		Delete(ctx); err != nil {
+		return err
+	}
```

Comment:
You should also delete the created secret that is referenced by the managed resource?!